website/docs/tutorials/nextjs.md

```diff
@@ -0,0 +1,313 @@
+
+The example below shows how you can setup Lingui to work with sever components in Next.js.
+
+In Next.js 13+ it is a common pattern to add a `[locale]` dir at the root of your app to facilitate i18n. This parameter will correspond the the language of the page visited by the user. e.g. english `en` and french `fr`. When we switch between languages we will redirect the user to the new language url instead of the client side approach of calling `i18n.activate(locale)`.
+
+```tsx
+// src/app/[locale]/layout.tsx
+import { t } from '@lingui/macro'
+import I18nProvider from '@/components/I18nProvider'
+import { loadCatalog, setI18n } from '@/utils/locales'
+
+type Params = {
+  locale: string
+}
+
+type Props = {
+  params: Params
+  children: React.ReactNode
+}
+
+export function generateMetadata({ params }: { params: Params }) {
+  const { locale } = params
+  const i18n = setI18n(locale)
+  return {
+    title: t(i18n)`Translation Demo`, // this is an example of a server only translation
+  }
+}
+
+export function generateStaticParams() {
+    return [{ locale: 'en' }, { locale: 'fr' }]
+}
+
+export default function RootLayout({ params, children }: Props) {
+    const { locale } = params
+    const messages = loadCatalog(locale)
+    setI18n(locale)
+    return (
+        <html lang={locale}>
+            <body>
+                <I18nProvider locale={locale} messages={messages}>
+                    {children}
+                </I18nProvider>
+            </body>
+        </html>
+    )
+}
+```
+
+You can use the Next.js middleware to [redirect to the default language](https://nextjs.org/docs/app/building-your-application/routing/internationalization) if the root `/` is visited.
+
+```ts
+// src/middleware.ts
+import { NextResponse, NextRequest } from "next/server"
+ 
+const locales = ['en', 'fr']
+ 
+export function middleware(req : NextRequest) {
+  const { pathname } = req.nextUrl
+  const pathnameHasLocale = locales.some(
+    (locale) => pathname.startsWith(`/${locale}/`) || pathname === `/${locale}`
+  )
+  if (pathnameHasLocale) return
+ 
+  const defaultLocale = 'en' // or find the "accept-language" in the header
+  req.nextUrl.pathname = `/${defaultLocale}${pathname}`
+  return NextResponse.redirect(req.nextUrl)
+}
+ 
+export const config = {
+  matcher: [
+    '/((?!_next).*)',
+  ],
+}
+```
+
+As you can see in the example above we slight modified the default `I18nProvider` with our own implementation. Here we pass the active message catalog as a prop. This is to avoid importing all catalogs inside the `I18nProvider` component itself and thus making them part of the client bundle. For the people who are wondering: Why we are not just passing an instance of `i18n` the original provider component? Server components can pass data to client components but the caveat is that this needs to be serializable data and `i18n` is not.
```

Comment:
if we will go with "@lingui/nextjs" may be pre-made this provider in that package?

Reply:
Hmm... not 100% sure. One the one hand you would want the package to "batteries included" but you don't want to inflate the scope to much.  Leaning more to "it should be included".